src/core/lib/iomgr/exec_ctx.h

```diff
@@ -80,10 +80,13 @@ namespace grpc_core {
  *  CONVENTIONS:
  *  - Instance of this must ALWAYS be constructed on the stack, never
  *    heap allocated.
- *  - Exactly one instance of ExecCtx must be created per thread. Instances must
- *    always be called exec_ctx.
  *  - Do not pass exec_ctx as a parameter to a function. Always access it using
  *    grpc_core::ExecCtx::Get().
+ *
+ * TODO(yashykt): Only allow one "active" ExecCtx on a thread at the same time.
```

Comment:
We should be moving to a mode where we only ever have one in existence at a given time.

Reply:
I think that's stage 3 (we just crossed stage 1 of separating ApplicationCallbackExecCtx from ExecCtx), but I can document that that's a goal.

Reply:
I've expanded this now.